### PR TITLE
Use control group target from mavlink message.

### PR DESCRIPTION
**Pull request by xgerrmann**
Previously the control group 0 was used for publishing the control targets from the incoming mavlink messages. In the mavlink specification, it is possible to set a specific control group. This ensures the correct control group is targeted.
This PR, accompanies the DroneCore PR: https://github.com/dronecore/DroneCore/pull/277

### CODE REVIEW

**Comment by dagar** [comment]
@xgerrmann please check the CI failure (trivial formatting).

**Comment by xgerrmann** [comment]
@dagar Can you look into this, please? 
If you agree with the implementation, I'll rebase on demand.

**Comment by dagar** [comment]
Looks correct, but don't we need more logic to handle group_mlx changing? There's nothing stopping you from sending the first set_actuator_control_target on group_mlx=0, which advertises and publishes actuator_controls_0, then the following message on group_mlx=1, but the publish will be broken.

**Comment by xgerrmann** [comment]
@dagar Thanks for the remark, haven't thought of it. The latest commit should solve this problem. One thing though, I have used a static variable, is this a correct implementation or should I define de variable in the header file off the mavlink receiver?

**Comment by xgerrmann** [comment]
@julianoes Might you have time to take a look at this?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -1055,18 +1055,44 @@ MavlinkReceiver::handle_message_set_actuator_control_target(mavlink_message_t *m
...
-
+			uint8_t control_group           = set_actuator_control_target.group_mlx;
+			static int8_t control_group_previous = -1;
 			/* Set duty cycles for the servos in actuator_controls_0 */
-			for (size_t i = 0; i < 8; i++) {
-				actuator_controls.control[i] = set_actuator_control_target.controls[i];
+			memcpy(actuator_controls.control, set_actuator_control_target.controls, 8 * sizeof(float));
+
+			/* Reset orb advertising when current control group is unequal to the previous control group */
+			if (!(control_group == control_group_previous)) {
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_receiver.cpp]: Why isn't this just `!=`?

Comment (julianoes) [reply src/modules/mavlink/mavlink_receiver.cpp]: And in this case, you should probably call `orb_unadvertise`.

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -1055,18 +1055,44 @@ MavlinkReceiver::handle_message_set_actuator_control_target(mavlink_message_t *m
 		if (_control_mode.flag_control_offboard_enabled) {
 
 			actuator_controls.timestamp = hrt_absolute_time();
-
+			uint8_t control_group           = set_actuator_control_target.group_mlx;
+			static int8_t control_group_previous = -1;
 			/* Set duty cycles for the servos in actuator_controls_0 */
-			for (size_t i = 0; i < 8; i++) {
-				actuator_controls.control[i] = set_actuator_control_target.controls[i];
+			memcpy(actuator_controls.control, set_actuator_control_target.controls, 8 * sizeof(float));
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_receiver.cpp]: The comment above is now a bit lonely. Either remove or move it.

**Comment by dagar** [comment]
Has this been tested?

**Comment by dagar** [comment]
Can you remove the extra commits and rebase on current master?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 3]
```diff
@@ -1055,18 +1055,47 @@ MavlinkReceiver::handle_message_set_actuator_control_target(mavlink_message_t *m
 		if (_control_mode.flag_control_offboard_enabled) {
 
 			actuator_controls.timestamp = hrt_absolute_time();
+			uint8_t control_group           = set_actuator_control_target.group_mlx;
+			static int8_t control_group_previous = -1;
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp]: Can you move the static into the class?

**Comment by AlexisTM** [comment]
NOTE: This has conflicts with https://github.com/PX4/Firmware/pull/7477 as it solves the same issue

**Comment by AlexisTM** [comment]
@xgerrmann #7477 is merged into the master; Thus this PR is already solved.
https://github.com/PX4/Firmware/blob/2250946eaaecd61a3a88e802b65537ede540f701/src/modules/mavlink/mavlink_receiver.cpp#L1120-L1165

**Comment by julianoes** [comment]
@dagar we should probably check if this can go in.

**Comment by julianoes** [comment]
At stalebot, the fight is on! :smile:

**Comment by AlexisTM** [comment]
This PR has been overseeded by #7477.

**Comment by julianoes** [comment]
Ok  thanks @AlexisTM

**Pull request closed without merging**